### PR TITLE
Modal popUp. Sort.

### PR DESCRIPTION
**Pull request by KarolisSm**
Possibility to sort vendors by types. Also added search in all vendors. Modal popUp for register and login views.

### CODE REVIEW

File: FromLocalsToLocals/Controllers/VendorsController.cs [hunk 1]
```diff
@@ -25,9 +31,28 @@ public VendorsController(AppDbContext context, UserManager<AppUser> userManager)
...
+        public async Task<IActionResult> AllVendors(string vendorType, string searchString)
         {
-            return View(await _context.Vendors.ToListAsync());
+            IQueryable<string> typeQuery = from m in _context.Vendors
+                                           orderby m.VendorTypeDb
+                                           select m.VendorTypeDb;
+
+            var vendors = from m in _context.Vendors
+                        select m;
+            if (!String.IsNullOrEmpty(searchString))
```
Comment (StauceLukas) [inline FromLocalsToLocals/Controllers/VendorsController.cs:42]: I think this should be !string.

File: FromLocalsToLocals/Views/Account/Register.cshtml [hunk 2]
```diff
@@ -4,10 +4,52 @@
...
+
+<div class="row">
+    <div class="col-md-12">
+        <form method="post">
+            <div class="modal fade" tabindex="-1" id="registerModal"
+                 data-keyboard="false" data-backdrop="static">
+                <div class="modal-dialog modal-lg">
+                    <div class="modal-content">
+                        <div class="modal-header">
+                            <h4 class="modal-title">Login</h4>
```
Comment (StauceLukas) [inline FromLocalsToLocals/Views/Account/Register.cshtml:16]: This bug when Register Modal have Login title.

**Review by StauceLukas — APPROVED** [review]
Great job on Modal implementation, our App became much more eye appealing.

**Review by Rangatin — APPROVED** [review]
I think that most errors were pointed out by Lukas, me myself didn't find more.

File: FromLocalsToLocals/Models/VendorTypeViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+using FromLocalsToLocals.Models;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using System.Collections.Generic;
+
+namespace MvcMovie.Models
```
Comment (HabaHaba05) [inline FromLocalsToLocals/Models/VendorTypeViewModel.cs:5]: I believe this namespace shouldn't be MvcMovie.

File: FromLocalsToLocals/Controllers/VendorsController.cs [hunk 4]
```diff
@@ -25,9 +31,28 @@ public VendorsController(AppDbContext context, UserManager<AppUser> userManager)
 
         [HttpGet]
         [AllowAnonymous]
-        public async Task<IActionResult> AllVendors()
+        public async Task<IActionResult> AllVendors(string vendorType, string searchString)
         {
-            return View(await _context.Vendors.ToListAsync());
+            IQueryable<string> typeQuery = from m in _context.Vendors
+                                           orderby m.VendorTypeDb
+                                           select m.VendorTypeDb;
```
Comment (HabaHaba05) [inline FromLocalsToLocals/Controllers/VendorsController.cs:38]: I believe there you are taking all types of vendors from the database by iterating each row. But if we have A LOT of vendors, I think it's inefficient. We have Enum in Utilities/Enums for vendor types, so you could use that.

But by your way, if we don't have for example "food" service created we won't get even the ability to select food services, which is good I think. 

So should we change or leave it as it is? I believe we should change it, but what @Rangatin and @StauceLukas  think about this?

File: FromLocalsToLocals/Views/Account/Register.cshtml (hunk 2 above)
Comment (KarolisSm) [reply FromLocalsToLocals/Views/Account/Register.cshtml:16]: Thank you:)

File: FromLocalsToLocals/Models/VendorTypeViewModel.cs (hunk 3 above)
Comment (KarolisSm) [reply FromLocalsToLocals/Models/VendorTypeViewModel.cs:5]: You are right.

**Review by HabaHaba05 — APPROVED** [review]

**Pull request merged**